backend/src/main/java/com/hackner/musicompass/discogsapi/service/DiscogsArtistApiService.java

```diff
@@ -19,23 +23,56 @@
     private final DiscogsApiEntityService discogsApiEntityService;
     private final String baseUrl = "https://api.discogs.com";
 
-
     @Autowired
     public DiscogsArtistApiService(RestTemplate restTemplate, DiscogsApiEntityService discogsApiEntityService) {
         this.restTemplate = restTemplate;
         this.discogsApiEntityService = discogsApiEntityService;
     }
 
-    public DiscogsArtistSearchResults getDiscogsArtistByName(String artistName) {
+    public DiscogsArtistSearchResults getDiscogsArtistListBySearchTerm(String searchTerm) {
 
-        String url = baseUrl + "/database/search?type=artist&q=" + artistName;
+        String url = baseUrl + "/database/search?type=artist&q=" + searchTerm;
 
         try {
             ResponseEntity<DiscogsArtistSearchResults> response =
                     restTemplate.exchange(url, HttpMethod.GET, discogsApiEntityService.createEntity(), DiscogsArtistSearchResults.class);
             return response.getBody();
         } catch (Exception e){
-            return DiscogsArtistSearchResults.builder().results(new DiscogsArtist[]{}).build();
+            return DiscogsArtistSearchResults.builder().results(Collections.<DiscogsArtist>emptyList()).build();
+        }
+    }
+
+    public List<DiscogsMasterRelease> getDiscogsMasterReleaseListByArtistName(String artistName) {
+
+        try {
+            ResponseEntity<DiscogsMasterReleaseSearchResults> response =
+                    restTemplate.exchange(dynamicUrl(artistName, 1),
+                            HttpMethod.GET,
+                            discogsApiEntityService.createEntity(),
+                            DiscogsMasterReleaseSearchResults.class);
+
+            List<DiscogsMasterRelease> releaseList = response.getBody().getResults();
```

Comment:
i would move the request to a separate mehtod with page parameter. Than you can use it here and inside of the for loop

Reply:
good point. done